transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/ListTransferDestinationActivity.java

```diff
@@ -0,0 +1,130 @@
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.hyperwallet.android.model.HyperwalletError;
+import com.hyperwallet.android.model.HyperwalletErrors;
+import com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod;
+import com.hyperwallet.android.ui.common.repository.Event;
+import com.hyperwallet.android.ui.common.view.error.DefaultErrorDialogFragment;
+import com.hyperwallet.android.ui.transfer.R;
+import com.hyperwallet.android.ui.transfer.viewmodel.ListTransferDestinationViewModel;
+import com.hyperwallet.android.ui.transfermethod.repository.TransferMethodRepositoryFactory;
+
+import java.util.List;
+
+public class ListTransferDestinationActivity extends AppCompatActivity {
+
+    public static final String EXTRA_SELECTED_DESTINATION = "SELECTED_DESTINATION";
+
+    private ListTransferDestinationViewModel mListTransferDestinationViewModel;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list_transfer_destination);
+
+        HyperwalletTransferMethod selectedDestination = getIntent().getParcelableExtra(EXTRA_SELECTED_DESTINATION);
+        if (selectedDestination == null) {
+            throw new IllegalArgumentException(
+                    "EXTRA_SELECTED_DESTINATION intent data is needed to start this activity");
+        }
+
+        mListTransferDestinationViewModel = ViewModelProviders.of(this,
+                new ListTransferDestinationViewModel.ListTransferDestinationViewModelFactory(
+                        TransferMethodRepositoryFactory.getInstance().getTransferMethodRepository()))
+                .get(ListTransferDestinationViewModel.class);
+        registerObservers();
+
+        if (savedInstanceState == null) {
+            initFragment(ListTransferDestinationFragment.newInstance(selectedDestination));
+        }
+    }
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        onBackPressed();
+        return true;
+    }
+
+    @Override
+    public void onBackPressed() {
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+        getWindow().setStatusBarColor(ContextCompat.getColor(this, R.color.colorPrimaryDark));
+        getWindow().getDecorView().setSystemUiVisibility(0);
+        super.onBackPressed();
+    }
+
+    private void initFragment(Fragment fragment) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        FragmentTransaction transaction = fragmentManager.beginTransaction();
+        transaction.add(R.id.list_destination_fragment, fragment);
+        transaction.commit();
+    }
+
+    private void registerObservers() {
```

Comment:
I think it's safer to check if the content hasn't been consumed in those events...

Reply:
yeah but we are disposing the UI after consuming so there's no chance for it to be consume again